### PR TITLE
feat: gitlab commit/mr data collecting

### PR DESCRIPTION
**Pull request by klesh**

### CODE REVIEW

**Comment by joncodo** [comment]
@klesh thanks. I need to look waaaay more into this. Did @UltimateBeaver assign you this work? I have already done this for projects and commits

**Comment by klesh** [comment]
Hi, @joncodo . Sorry for the inconvenience, it's not my intent to overlap your work.
I need gitlab data for some cross-data-source indicators, and fast. 
Task is due this week.
Please take a look if all possible, many thanks

File: src/plugins/gitlab-pond/src/collector/fetcher.js [hunk 1]
```diff
@@ -1,32 +1,43 @@
...
+      }
+    }
+    if (res.data && res.data.message) {
+      throw new Error(`status: ${res.status} messgae: ${res.data.message}`)
+    }
+    console.log(`[gitlab] Fetched data from ${resourceUri}`)
+    return res
+  },
+
+  async * fetchPaged (apiPath, maxRetry = 3, startAt = 0, pageSize = 100) {
```
Comment (UltimateBeaver) [inline src/plugins/gitlab-pond/src/collector/fetcher.js:30]: Better to name the first parameter as `resourceUri`?

**Comment by UltimateBeaver** [comment]
Looking good, merging now.

**Pull request merged**